### PR TITLE
fill in some key concepts

### PR DESCRIPTION
**Pull request by greyscaled**

### CODE REVIEW

File: introduction/key-concepts-in-glaredb.md [hunk 1]
```diff
@@ -1,20 +1,98 @@
...
 ---
 
-## Key Concepts in GlareDB
+# Key Concepts in GlareDB
 
-### Table Functions
+## Table Functions
 
-### External Data Sources
+A core tenant of GlareDB is that [All Data are SQL Addressable], even data not
```
Comment (talagluck) [inline introduction/key-concepts-in-glaredb.md:12]:
```suggestion
A core tenet of GlareDB is that [All Data are SQL Addressable], even data not
```

**Review by talagluck — COMMENTED** [review]
What's here so far looks good!

File: introduction/key-concepts-in-glaredb.md [hunk 2]
```diff
@@ -1,20 +1,148 @@
...
+
+- Querying local, scratch or exploratory data
+- One-offs or infrequent queries
+
+and add external data sources for the remaining cases.
+
+## Native Data Storage
+
+In addition to reading files and external data sources, GlareDB has native
+storage. Tables can be created and data can be inserted, updated and delete as
```
Comment (greyscaled) [inline introduction/key-concepts-in-glaredb.md:92]:
```suggestion
storage. Tables can be created and data can be inserted, updated, and deleted as
```

File: introduction/key-concepts-in-glaredb.md [hunk 3]
```diff
@@ -1,20 +1,148 @@
...
+in the cloud can be allocated for units of work, meaning that queries can be
+partitioned and distributed in parallel.
+
+By designing the system this way, our billing system is consumption-based and
+transparent. We bill for bytes processed and bytes stored, rather than resource
+or time-based metrics. Billing on these metrics means that queries can get
+faster, spanning multiple worker nodes and various compute resources without
+the end-user incurring any change in cost!
+
+For more information, read our [announcement post on GlareDB Pro]
```
Comment (greyscaled) [inline introduction/key-concepts-in-glaredb.md:141]:
```suggestion
For more information, read our [announcement post on GlareDB Pro].
```

File: introduction/key-concepts-in-glaredb.md [hunk 4]
```diff
@@ -53,13 +53,12 @@ INNER JOIN read_csv('./my_csv') c ON m.id = c.id;
...
+sources and loaded into an analytics warehouse. An underlying assumption is
 analytics should own storage and infrastructure, often at great complexity.
 
-GlareDB shifts the dynamic by recognizing that data is accessible where it is,
-for example with table functions. However, it is often the case that the same
-sources are accessed frequently and benefit from being recognized as known
-sources.
+GlareDB shifts the dynamic by recognizing data is accessible where it is.
+However, it is often the case that the same sources are accessed frequently and
+benefit from being recognized as known sources.
```
Comment (tychoish) [inline introduction/key-concepts-in-glaredb.md:61]:
```suggestion
GlareDB provides access to data where it is, without requiring a move.
Even though data is more dynamic, having data sources configured in GlareDB 
makes it easier to access common data sources and share configuration between
users.
```

File: introduction/key-concepts-in-glaredb.md [hunk 5]
```diff
@@ -89,10 +88,54 @@ and add external data sources for the remaining cases.
 
 ## Native Data Storage
 
+In addition to reading files and external data sources, GlareDB has native
+storage. Tables can be created and data can be inserted, updated and delete as
```
Comment (tychoish) [inline introduction/key-concepts-in-glaredb.md]:
```suggestion
In addition to reading files and external data sources, GlareDB its own (native)
storage. You can create tables, and then insert, update, and delete data as needed, as in:
```

File: introduction/key-concepts-in-glaredb.md [hunk 6]
```diff
@@ -89,10 +88,54 @@ and add external data sources for the remaining cases.
...
+When connecting via the Postgres protocol, all execution occurs in GlareDB. For
+example if using [GlareDB Cloud], queries will not be hybrid: all execution
+occurs in the cloud.
+
 ## Serverless Native
 
+GlareDB has always been designed to be serverless. Storage and compute are
+independently scalable and compute nodes are multi-tenant. Any GlareDB process
+in the cloud can be allocated for units of work, meaning that queries can be
+partitioned and distributed in parallel.
```
Comment (tychoish) [inline introduction/key-concepts-in-glaredb.md:133]:
```suggestion
GlareDB is built from the bottom up with serverless operation and delivery in mind.
Storage and compute are separated and can be scaled independently, and all billing 
is based on data processed (not time!) so users only pay for the resources they use.
```

**Pull request merged**